tests/test_sim_results/electric_blue/test_volume_timeseries.py

```diff
@@ -0,0 +1,203 @@
+            (DateTime(2020, 1, 1, 2, 0), DateTime(2020, 1, 1, 2, 30)),
+        ], resolution=AggregationResolution.RES_1_HOUR)
+
+    def test_adapt_time_slot_for_1_week_resolution(self):
+        self.check_time_slot_pairs([
+            (DateTime(2020, 1, 1, 0, 0).start_of("week"), DateTime(2020, 1, 1, 0, 0)),
+            (DateTime(2020, 2, 1, 0, 20).start_of("week"), DateTime(2020, 2, 1, 0, 20)),
+        ], resolution=AggregationResolution.RES_1_WEEK)
+
+    def test_adapt_time_slot_for_1_month_resolution(self):
+        self.check_time_slot_pairs([
+            (DateTime(2020, 1, 1, 0, 0).start_of("month"), DateTime(2020, 1, 1, 0, 0)),
+            (DateTime(2020, 2, 29, 0, 20).start_of("month"), DateTime(2020, 2, 29, 0, 20)),
+        ], resolution=AggregationResolution.RES_1_MONTH)
+
+    def test_adapt_time_slot_for_1_year_resolution(self):
+        self.check_time_slot_pairs([
+            (DateTime(2020, 1, 1, 0, 0).start_of("month"), DateTime(2020, 1, 1, 0, 0)),
+            (DateTime(2020, 2, 29, 0, 20).start_of("year"), DateTime(2020, 2, 29, 0, 20)),
+        ], resolution=AggregationResolution.RES_1_YEAR)
+
+    def test_add_asset_time_series_for_1_year_resolution(self, forward_time_series):
+        volume_time_series = AssetVolumeTimeSeries(
+            "UUID", 5, resolution=AggregationResolution.RES_1_YEAR)
+
+        for market_type, asset_stats in forward_time_series:
+            volume_time_series.add(asset_stats, market_type)
+
+        self.check_time_slots_year(volume_time_series)
+        assert volume_time_series._asset_volume_time_series_buffer == {
+            DateTime(2020, 1, 1, 0, 0, 0): {
+                DateTime(2020, 1, 1, 0, 0, 0): {
+                    "SSP": 36164.701938689024,
+                    "YEAR_FORWARD": {
+                        "bought_kWh": sum_of_energy(
+                            DateTime(2020, 1, 1, 0, 0, 0),
+                            DateTime(2020, 1, 1, 0, 0, 0).end_of("year"),
+                            1
+                        ),
+                        "sold_kWh": sum_of_energy(
+                            DateTime(2020, 1, 1, 0, 0, 0),
+                            DateTime(2020, 1, 1, 0, 0, 0).end_of("year"),
+                            1
+                        )},
+                    "MONTH_FORWARD": {
+                        "bought_kWh": sum_of_energy(
+                            DateTime(2020, 1, 1, 0, 0, 0),
+                            DateTime(2020, 1, 1, 0, 0, 0).end_of("month"),
+                            1
+                        ),
+                        "sold_kWh": sum_of_energy(
+                            DateTime(2020, 1, 1, 0, 0, 0),
+                            DateTime(2020, 1, 1, 0, 0, 0).end_of("month"),
+                            1
+                        )
+                    },
+                    "WEEK_FORWARD": {
+                        "bought_kWh": sum_of_energy(
```

Comment:
So, do I get it right: This is the sum of energy that has been bought in the WEEK_FORWARD markets for the whole year?

Reply:
Isn't the user interested in a profile of the weekly markets?

Reply:
This is the yearly resolution of the profile. If one is interested in a weekly resolution, they shall check another graph!

Reply:
Ahhhh, I think this answer made it all clear to me. let me re-review.